src/main/java/ui/components/pickers/PickerLabel.java

```diff
@@ -1,25 +1,32 @@
 package ui.components.pickers;
 
+import java.util.Optional;
+
 import backend.resource.TurboLabel;
+
 import com.sun.javafx.tk.FontLoader;
 import com.sun.javafx.tk.Toolkit;
+
 import javafx.scene.Node;
 import javafx.scene.control.Label;
 import javafx.scene.control.Tooltip;
 
-// for use with LabelPickerDialog
+/**
+ * This class is to represent a label in LabelPickerDialog
+ *
+ * It contains attributes such as selected, highlighted, removed and faded in order
+ * to produce the appropriate styled node through getNode()
+ */
```

Comment:
Maybe we could split `isTop` into two fields, one for whether to show the full name, another for whether to show the tick? Where the label is going to be placed shouldn't leak into this class.